core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java

```diff
@@ -145,8 +204,141 @@ public int deleteRoleMetasByLegacyTimeLine(long legacyTimeLine, int limit) {
+      return MetalakeMetaService.getInstance().getMetalakeIdByName(fullName);
+    }
+
+    List<String> names = DOT.splitToList(fullName);
+    long catalogId =
+        CatalogMetaService.getInstance().getCatalogIdByMetalakeIdAndName(metalakeId, names.get(0));
+    if (type == MetadataObject.Type.CATALOG) {
+      return catalogId;
+    }
+
+    long schemaId =
+        SchemaMetaService.getInstance().getSchemaIdByCatalogIdAndName(catalogId, names.get(1));
+    if (type == MetadataObject.Type.SCHEMA) {
+      return schemaId;
+    }
+
+    if (type == MetadataObject.Type.FILESET) {
+      return FilesetMetaService.getInstance().getFilesetIdBySchemaIdAndName(schemaId, names.get(2));
+    } else if (type == MetadataObject.Type.TOPIC) {
+      return TopicMetaService.getInstance().getTopicIdBySchemaIdAndName(schemaId, names.get(2));
+    } else if (type == MetadataObject.Type.TABLE) {
+      return TableMetaService.getInstance().getTableIdBySchemaIdAndName(schemaId, names.get(2));
+    }
+
+    throw new IllegalArgumentException(String.format("Doesn't support the type %s", type));
+  }
+
+  String getSecurableObjectFullName(SecurableObjectPO securableObjectPO) {
+    if (securableObjectPO.getType().equals(ROOT)) {
+      return ALL_METALAKES;
+    }
 
-    return roleDeletedCount[0] + userRoleRelDeletedCount[0] + groupRoleRelDeletedCount[0];
+    MetadataObject.Type type = MetadataObject.Type.valueOf(securableObjectPO.getType());
+    if (type == MetadataObject.Type.METALAKE) {
+      MetalakePO metalakePO =
+          MetalakeMetaService.getInstance().getMetalakePOById(securableObjectPO.getEntityId());
+      if (metalakePO == null) {
+        return null;
+      }
+
+      return metalakePO.getMetalakeName();
+    }
+
+    if (type == MetadataObject.Type.CATALOG) {
+      return getCatalogFullName(securableObjectPO.getEntityId());
+    }
+
+    if (type == MetadataObject.Type.SCHEMA) {
+      return getSchemaFullName(securableObjectPO.getEntityId());
+    }
+
+    if (type == MetadataObject.Type.TABLE) {
+      TablePO tablePO =
+          TableMetaService.getInstance().getTablePOById(securableObjectPO.getEntityId());
+      if (tablePO == null) {
+        return null;
+      }
+      return getSchemaFullName(tablePO.getSchemaId()) + "." + tablePO.getTableName();
+    }
+
+    if (type == MetadataObject.Type.TOPIC) {
+      TopicPO topicPO =
+          TopicMetaService.getInstance().getTopicPOById(securableObjectPO.getEntityId());
+      if (topicPO == null) {
+        return null;
+      }
+      return getSchemaFullName(topicPO.getSchemaId()) + "." + topicPO.getTopicName();
+    }
+
+    if (type == MetadataObject.Type.FILESET) {
+      FilesetPO filesetPO =
+          FilesetMetaService.getInstance().getFilesetPOById(securableObjectPO.getEntityId());
+      if (filesetPO == null) {
+        return null;
+      }
+      return getSchemaFullName(filesetPO.getSchemaId()) + "." + filesetPO.getFilesetName();
+    }
+
+    return null;
```

Comment:
I noticed that `getSecurableObjectEntityId` will throw an exception if the type is not supported, do you have any considerations here?

Reply:
Done.